File: typescript/Behavioral/15-Iterator/Model/JobCard.ts
const shortId = () => Math.random().toString(36).slice(2, 10);

export class JobCard {
  readonly jobId: string = shortId();
  status: string = "Queued";

  constructor(
    readonly title: string,
    readonly category: string,
    readonly priority: number = 0,
  ) {}

  toString(): string {
    return `[P${this.priority}] #${this.jobId} '${this.title}' [${this.category}] — ${this.status}`;
  }
}

File: csharp/Behavioral/15-Iterator/Model/JobCard.cs
public class JobCard
{
    public string JobId    { get; init; } = Guid.NewGuid().ToString()[..8];
    public string Title    { get; init; } = string.Empty;
    public string Category { get; init; } = string.Empty;
    public int    Priority { get; init; } = 0;
    public string Status   { get; set;  } = "Queued";

    public override string ToString() =>
        $"[P{Priority}] #{JobId} '{Title}' [{Category}] — {Status}";
}
